src/librustc_errors/lib.rs

```diff
@@ -652,14 +692,16 @@ impl Handler {
     }
 
     pub fn err_count(&self) -> usize {
-        self.err_count.load(SeqCst)
+        self.err_count.load(SeqCst) + self.stashed_diagnostics.borrow().len()
```

Comment:
Another approach would be to bump the count when stashing and decrement when stealing; we'd need to add `fetch_sub` to `Atomic<T>`.

Reply:
This will need a rebase after #64272 gets merged.